custom_components/kia_uvo/coordinator.py

```diff
@@ -159,63 +160,83 @@ async def async_check_and_refresh_token(self):
             self.vehicle_manager.check_and_refresh_token
         )
 
+    async def async_wait_for_action_completed(
+        self, vehicle_id: int, action_id: str, interval: int = 5, max_count: int = 6
+    ):
+        for count in range(1, max_count + 1):
+            _LOGGER.debug(f"Last action check: waiting {interval} seconds")
+            await asyncio.sleep(interval)
```

Comment:
I don't have time to check right now but I believe home assistant discourages this as it ties up the event loop.  They have a specific command that you use to have it schedule something X time in the future instead.

Reply:
Thank you, will see if I can figure out the correct way to do it

Reply:
Okay, so I gave this some thought. I wonder if the polling logic belongs in the API rather than the coordinator? Given that we support checking status for multiple vehicles simultaneously, and polling logic may be API-dependent.

Not sure this is the best way to go about it, but we could add a callback to vehicle_manager.check_action_status that gets executed when the action is completed. In the coordinator we could then have this callback function refresh home assistant data, handle errors etc.

On the API side we could poll using asyncio.call_later for or something similar to avoid locking the event loop.

Would appreciate your thoughts on this!